opal/mca/pmix/pmix2x/Makefile.am

```diff
@@ -21,6 +23,12 @@ sources = \
         pmix2x_client.c \
         pmix2x_server_south.c \
         pmix2x_server_north.c
+else
+SUBDISR =
+
+endif
+
+DIST_SUBDIRS = $(SUBDIRS)
```

Comment:
`DIST_SUBDIRS` should never be empty; it always needs to contain all the subdirs, and according to the code above, `SUBDIRS` may be empty.  The same is true for the other edits in this PR.

I know you guys didn't know about https://github.com/open-mpi/ompi/pull/1687 before making this PR, but I think I might have said something analogous over there (and then let that PR fall off my radar...)  Ah yes, here it is: https://github.com/open-mpi/ompi/pull/1687#issuecomment-222313382

I'll go have a close look at #1687 now...